cookbook/console/style.rst

```diff
@@ -0,0 +1,363 @@
+    It displays the given string or array of strings highlighted as a successful
+    message (with a green background and the ``[OK]`` label). It's meant to be
+    used once to display the final result of executing the given command, but you
+    can use it repeatedly during the execution of the command::
+
+        // use simple strings for short success messages
+        $io->success('Lorem ipsum dolor sit amet');
+
+        // ...
+
+        // consider using arrays when displaying long success messages
+        $io->success(array(
+            'Lorem ipsum dolor sit amet',
+            'Consectetur adipiscing elit',
+        ));
+
+:method:`Symfony\\Component\\Console\\Helper\\StyleInterface::warning`
+    It displays the given string or array of strings highlighted as a warning
+    message (with a read background and the ``[WARNING]`` label). It's meant to be
+    used once to display the final result of executing the given command, but you
+    can use it repeatedly during the execution of the command::
+
+        // use simple strings for short warning messages
+        $io->warning('Lorem ipsum dolor sit amet');
+
+        // ...
+
+        // consider using arrays when displaying long warning messages
+        $io->warning(array(
+            'Lorem ipsum dolor sit amet',
+            'Consectetur adipiscing elit',
+        ));
+
+:method:`Symfony\\Component\\Console\\Helper\\StyleInterface::error`
+    It displays the given string or array of strings highlighted as an error
+    message (with a read background and the ``[ERROR]`` label). It's meant to be
+    used once to display the final result of executing the given command, but you
+    can use it repeatedly during the execution of the command::
+
+        // use simple strings for short error messages
+        $io->error('Lorem ipsum dolor sit amet');
+
+        // ...
+
+        // consider using arrays when displaying long error messages
+        $io->error(array(
+            'Lorem ipsum dolor sit amet',
+            'Consectetur adipiscing elit',
+        ));
+
+Defining your Own Styles
+------------------------
+
+If you don't like the design of the commands that use the Symfony Style, you can
+define your own set of console styles. Just create a class that implements the
+:class:`Symfony\Component\Console\Style\StyleInterface`::
+
+    namespace AppBundle\Console;
+
+    use Symfony\Component\Console\Style\StyleInterface;
+
+    class CustomStyle
+    {
+        // implement the methods of the interface...
+    }
+
+Then, instantiate this custom class instead of the default ``SymfonyStyle`` in
+your commands. Thanks to the ``StyleInterface`` you won't need to change the code
+of your commands to change their appearance::
+
+    namespace AppBundle\Console;
+
+    use AppBundle\Console\CustomStyle;
+    use Symfony\Component\Console\Style\SymfonyStyle;
+
+    public function execute()
+    {
+        // Before
+        // $io = new SymfonyStyle();
```

Comment:
Missing `$input` and `$output` arguments ? I think there are mandatory to understand there is no black magic behind this.